### DISCUSSION TITLE
Toggle button for 'make subresource unique'

### DISCUSSION THREAD

**Discussion by AndHalfAFish** [godotengine/godot-proposals#4293] [Editor]
I often find myself copy/pasting an area node with child collisionshape within my scene because it requires less mouse clicks than adding 2 nodes for every hotspot I need. I often forget to make the collisionshape a 'unique resource' (because the option is hidden in a submenu) which results in me inadvertently changing the dimensions of a previously arranged hotspot.

It would be much more convenient to have a checkbox / toggle button : less clicks + easy to check at a glance if the option was enabled. I now have no way of knowing if the option was enabled, so I find myself clicking that option often ... just in case.

![unique_subresource](https://user-images.githubusercontent.com/59080934/160290537-3dc2d550-c3b4-4b9c-a0ad-cc71dc24cbe3.png)

**Comment by Calinou** [comment]
How would undoing **Make Sub-Resources Unique** work if no undo/redo history is available? I don't think the editor is able to know _which_ subresources should be made shared again.

**Comment by theraot** [comment]
How about "paste" and "paste duplicating resources"?

**Reply by AndHalfAFish** [reply]
@Calinou  : Oh, I haven't considered the 'undo' option (I don't need that) - maybe toggle button wasn't a good visual representation.
I just wanted to say that I would prefer it if that option wasn't hidden, and if I could clearly see whether I have already made my resource unique.
I nearly always need my collisionshapes to be unique ... so, a bit of a hassle to use it as it is now.

@theraot : for me personally, that wouldn't be helpful. I would like it to be visually obvious that a resource is unique.

**Comment by lentsius-bark** [comment]
Since we're talking about resources and subresources here, it would be incredible to have an indicator of how many objects are using the resource.
Sometimes I am just not sure whether I should be making it unique or not, and a number indicator would be especially helpful when making something like a particle process resource unique, and forgetting about all of its sub resources. 
Let me put together a quick mockup :)

**Reply by Calinou** [reply]
> Since we're talking about resources and subresources here, it would be incredible to have an indicator of how many objects are using the resource.

This is already being tracked in https://github.com/godotengine/godot-proposals/issues/904 and https://github.com/godotengine/godot-proposals/issues/800.
